### PR TITLE
Make release v23.5

### PR DESCRIPTION
**Pull request by brosaplanella**
**WAIT TO MERGE UNTIL WE ARE HAPPY WITH THE RELEASE CANDIDATE**

I wasn't sure whether to have v23.5 or v23.5rc0 in the CHANGELOG and the CITATION. Everywhere else it is `23.5rc0` which should make a pre-release in PyPI.

### CODE REVIEW

**Comment by tinosulzer** [comment]
23.5 in the changelog is good, but we need to change the date

**Comment by brosaplanella** [comment]
> 23.5 in the changelog is good, but we need to change the date

Fair point. I have updated the date now, I will try to make the release and publish to pypi, but I am not sure if I have the permissions to trigger the action and actually push to pypi. I will let you know if not and you can trigger it.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,33 @@
 # [Unreleased](https://github.com/pybamm-team/PyBaMM/)
 
+# [v23.5](https://github.com/pybamm-team/PyBaMM/tree/v23.5) - 2023-06-18
```
Comment (Saransh-cpp) [inline CHANGELOG.md:3]: Will the date here be updated?

**Review by Saransh-cpp — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (brosaplanella) [reply CHANGELOG.md:3]: I was thinking of keeping it as it is the date of the feature freeze

**Pull request merged**